collect_app/src/main/java/org/odk/collect/android/activities/InstanceUploaderActivity.java

```diff
@@ -210,23 +212,40 @@ public void uploadingComplete(HashMap<String, String> result) {
             // tried to close a dialog not open. don't care.
         }
 
-        StringBuilder selection = new StringBuilder();
+        StringBuilder selection = null;
         Set<String> keys = result.keySet();
         Iterator<String> it = keys.iterator();
 
-        String[] selectionArgs = new String[keys.size()];
-        int i = 0;
-        while (it.hasNext()) {
-            String id = it.next();
-            selection.append(InstanceColumns._ID + "=?");
-            selectionArgs[i++] = id;
-            if (i != keys.size()) {
-                selection.append(" or ");
+        String[] selectionArgs = null;
+        StringBuilder message = new StringBuilder();
+        int count = keys.size();
+
+        while (count != 0) {
```

Comment:
As @nribeka pointed out, this would conceptually be better as a test on `> 0`. It captures the scenario a bit better and ensures that if ever `count` accidentally became negative, the loop would still stop.